bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -466,80 +487,141 @@ public void start() {
+                        LOG.error("Interrupted while running periodic check", ie);
+                    } catch (BKException bke) {
+                        LOG.error("Exception running periodic check", bke);
+                    } catch (IOException ioe) {
+                        LOG.error("I/O exception running periodic check", ioe);
+                    } catch (ReplicationException.UnavailableException ue) {
+                        LOG.error("Underreplication manager unavailable running periodic check", ue);
                     }
-                    }, initialDelay, interval, TimeUnit.SECONDS);
+                }
+                }, initialDelay, interval, TimeUnit.SECONDS);
+        } else {
+            LOG.info("Periodic checking disabled");
+        }
+    }
+
+    private void scheduleMetadataCheckTask(){
+        long interval = conf.getAuditorPeriodicMetadataCheckInterval();
+
+        if (interval > 0) {
+            LOG.info("Auditor periodic metadata check enabled" + " 'auditorPeriodicMetadataCheckInterval' {} seconds",
+                    interval);
+
+            long metadataCheckLastExecutedCTime;
+            long durationSinceLastExecutionInSecs;
+            long initialDelay;
+            try {
+                metadataCheckLastExecutedCTime = ledgerUnderreplicationManager.getMetadataCheckCTime();
+            } catch (UnavailableException ue) {
+                LOG.error("Got UnavailableException while trying to get metadataCheckCTime", ue);
+                metadataCheckLastExecutedCTime = -1;
+            }
+            if (metadataCheckLastExecutedCTime == -1) {
+                durationSinceLastExecutionInSecs = -1;
+                initialDelay = 0;
             } else {
-                LOG.info("Periodic checking disabled");
+                durationSinceLastExecutionInSecs = (System.currentTimeMillis() - metadataCheckLastExecutedCTime)
+                        / 1000;
+                if (durationSinceLastExecutionInSecs < 0) {
+                    // this can happen if there is no strict time ordering
+                    durationSinceLastExecutionInSecs = 0;
+                }
+                initialDelay = durationSinceLastExecutionInSecs > interval ? 0
+                        : (interval - durationSinceLastExecutionInSecs);
             }
+            LOG.info(
+                    "metadataCheck scheduling info.  metadataCheckLastExecutedCTime: {} "
+                            + "durationSinceLastExecutionInSecs: {} initialDelay: {} interval: {}",
+                            metadataCheckLastExecutedCTime, durationSinceLastExecutionInSecs, initialDelay, interval);
+
+            executor.scheduleAtFixedRate(new Runnable() {
+                public void run() {
+                    try {
+                        Stopwatch stopwatch = Stopwatch.createStarted();
+                        LOG.info("Starting MetadataCheck");
+                        metadataCheck();
```

Comment:
How do you know if the previous instance of the metadataCheck is running or not? I guess we need to make sure that we have only one instance running at any given time.